Zend/zend_language_parser.y

```diff
@@ -63,6 +63,7 @@ static YYSIZE_T zend_yytnamerr(char*, const char*);
 %precedence '=' T_PLUS_EQUAL T_MINUS_EQUAL T_MUL_EQUAL T_DIV_EQUAL T_CONCAT_EQUAL T_MOD_EQUAL T_AND_EQUAL T_OR_EQUAL T_XOR_EQUAL T_SL_EQUAL T_SR_EQUAL T_POW_EQUAL T_COALESCE_EQUAL
 %left '?' ':'
 %right T_COALESCE
+%left T_PIPE
```

Comment:
This appears to be binding T_PIPE before T_COALESCE, whereas I believe it should bind lower.  Moving it up a line results in a seemingly unrelated Phar error.  Suggestions welcome on what I'm doing wrong here.

Reply:
What is the phar error you get?

Reply:
```
PEAR package PHP_Archive not installed: generated phar will require PHP's phar extension be enabled.

Fatal error: Uncaught Error: Value of type null is not callable in /home/crell/Projects/php-src/ext/phar/phar.php:1057
Stack trace:
#0 /home/crell/Projects/php-src/ext/phar/phar.php(225): PharCommand->cli_cmd_run_pack(Array)
#1 /home/crell/Projects/php-src/ext/phar/phar.php(2089): CLICommand->__construct(19, Array)
#2 {main}
  thrown in /home/crell/Projects/php-src/ext/phar/phar.php on line 1057
make: *** [Makefile:441: ext/phar/phar.phar] Error 255
```

That error does not appear with the code as written in the patch right now.

I would expect these to parse in lexical order, so that the current code would bind ?? before |>, but that's not the behavior I'm seeing.  (I have a not-yet-committed test for that; waiting for it to be passing before adding it.)

Reply:
The associativity list defines [increasing precedence order](https://www.gnu.org/software/bison/manual/html_node/Using-Precedence.html) (as can be seen below: first lower-precedence `%left '+' '-'`, then higher-precedence `%left '*' '/' '%'`). But that doesn't explain why you get a Phar error if you swap the lines...